pkg/util/signal.go

```diff
@@ -7,32 +7,37 @@ import (
 	"sync"
 )
 
-func NewSingleSignalPair() (SignalSender, SignalReceiver) {
-	sigCh := make(chan struct{})
+func NewSingleSignalPair[T any]() (SignalSender[T], SignalReceiver[T]) {
+	sigCh := make(chan T)
```

Comment:
```suggestion
	sigCh := make(chan T, 1)
```

If this has a buffer size of 0, then the send will block until the message is received.

Reply:
Ahh right. Definitely don't want to block.